WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetUpdater.kt

```diff
@@ -5,69 +5,65 @@ import android.appwidget.AppWidgetManager
-import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureFragment.ViewType.WEEK_VIEWS
 import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color.DARK
 import org.wordpress.android.ui.stats.refresh.lists.widget.StatsWidgetConfigureViewModel.Color.LIGHT
 import org.wordpress.android.util.NetworkUtilsWrapper
-import org.wordpress.android.util.image.ImageManager
-import org.wordpress.android.util.image.ImageType.ICON
 import org.wordpress.android.viewmodel.ResourceProvider
 import javax.inject.Inject
-import kotlin.random.Random
-
-private const val MIN_WIDTH = 250
 
 class ViewsWidgetUpdater
 @Inject constructor(
     private val appPrefsWrapper: AppPrefsWrapper,
     private val siteStore: SiteStore,
-    private val imageManager: ImageManager,
     private val networkUtilsWrapper: NetworkUtilsWrapper,
-    private val resourceProvider: ResourceProvider
+    private val resourceProvider: ResourceProvider,
+    private val widgetUtils: WidgetUtils
 ) : WidgetUpdater {
     override fun updateAppWidget(
         context: Context,
         appWidgetManager: AppWidgetManager,
         appWidgetId: Int
     ) {
-        val minWidth = appWidgetManager.getAppWidgetOptions(appWidgetId)
-                .getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_WIDTH, 300)
-        val showChangeColumn = minWidth > MIN_WIDTH
+        val showChangeColumn = widgetUtils.isWidgetWiderThanLimit(appWidgetManager, appWidgetId)
         val colorMode = appPrefsWrapper.getAppWidgetColor(appWidgetId) ?: LIGHT
         val siteId = appPrefsWrapper.getAppWidgetSiteId(appWidgetId)
         val siteModel = siteStore.getSiteBySiteId(siteId)
         val networkAvailable = networkUtilsWrapper.isNetworkAvailable()
         val layout = when (colorMode) {
-            DARK -> R.layout.stats_widget_views_dark
-            LIGHT -> R.layout.stats_widget_views_light
+            DARK -> R.layout.stats_widget_list_dark
+            LIGHT -> R.layout.stats_widget_list_light
         }
         val views = RemoteViews(context.packageName, layout)
-        val siteIconUrl = siteModel?.iconUrl
-        val awt = AppWidgetTarget(context, R.id.widget_site_icon, views, appWidgetId)
-        imageManager.load(awt, context, ICON, siteIconUrl ?: "", FIT_START)
+        views.setTextViewText(R.id.widget_title, resourceProvider.getString(R.string.stats_views))
+        widgetUtils.setSiteIcon(siteModel, context, views, appWidgetId)
         siteModel?.let {
-            views.setOnClickPendingIntent(R.id.widget_title, getPendingSelfIntent(context, siteModel.id))
+            views.setOnClickPendingIntent(
+                    R.id.widget_title,
+                    widgetUtils.getPendingSelfIntent(context, siteModel.id, DAY)
+            )
             views.setPendingIntentTemplate(R.id.widget_content, getPendingTemplate(context))
```

Comment:
We are calling setPendingIntentTemplate on `widget_content` in `widgetUtils.showList` below, so I wonder is it ok to remove it here?

Reply:
good catch 👍